### PR TITLE
Allow to configure the property of SymfonyUserProvider

### PR DESCRIPTION
**Pull request by notFloran**

### CODE REVIEW

File: src/Provider/User/SymfonyUserProvider.php [hunk 1]
```diff
@@ -10,21 +12,33 @@ final class SymfonyUserProvider implements UserProviderInterface
...
+    private $propertyName;
+
+    public function __construct(TokenStorageInterface $tokenStorage = null, PropertyAccessorInterface $propertyAccessor = null)
     {
         $this->tokenStorage = $tokenStorage;
+        $this->propertyAccessor = $propertyAccessor ?: PropertyAccess::createPropertyAccessor();
+        $this->propertyName = 'username';
     }
 
     public function getUser(): string
```
Comment (nclsHart) [inline src/Provider/User/SymfonyUserProvider.php:44]: What do you think about replacing the setter by an optional argument `propertyName` on `getUser`:

```php
public function getUser(string $propertyName = 'username'): string;
```

**Review by nclsHart — CHANGES_REQUESTED** [review]
> The default provider returns:
> * anonymous when no user is logged
> * the username of the current logged user
> * cli

The doc should be updated accordingly to this new feature.

**Comment by notFloran** [comment]
The PR is not over, I would like to come up with something like this:

```yaml
the_red_dot_monolog_extra:
    provider:
        user: 
            service_id: your_own_provider_service_id
            property_name: email
```

or

```yaml
the_red_dot_monolog_extra:
    processor:
        user:
            enabled: true
            property_name: email
```

File: README.md [hunk 2]
```diff
@@ -75,11 +75,19 @@ the_red_dot_monolog_extra:
         user: true
 ```
 
-The default provider returns:
+The default provider `SymfonyUserProvider` returns:
 * anonymous when no user is logged
 * the username of the current logged user
 * cli
 
+You can customize the provider to replace the username by another property :
```
Comment (nclsHart) [inline README.md]:
```suggestion
You can customize the provider to replace the username by another property:
```

File: src/Provider/User/SymfonyUserProvider.php [hunk 3]
```diff
@@ -10,21 +12,33 @@ final class SymfonyUserProvider implements UserProviderInterface
     /**
      * User for anonymous.
      */
-    const USER_ANONYMOUS = 'anonymous';
+    public const USER_ANONYMOUS = 'anonymous';
```
Comment (nclsHart) [inline src/Provider/User/SymfonyUserProvider.php]: private?

File: src/Provider/User/SymfonyUserProvider.php [hunk 4]
```diff
@@ -10,21 +12,33 @@ final class SymfonyUserProvider implements UserProviderInterface
...
 
     /**
      * Value for user when we are in cli.
      */
-    const USER_CLI = 'cli';
+    public const USER_CLI = 'cli';
```
Comment (nclsHart) [inline src/Provider/User/SymfonyUserProvider.php]: private?

File: src/Provider/User/SymfonyUserProvider.php [hunk 5]
```diff
@@ -10,21 +12,33 @@ final class SymfonyUserProvider implements UserProviderInterface
...
+     * @var PropertyAccessorInterface
+     */
+    private $propertyAccessor;
+
+    /**
+     * @var string
+     */
+    private $propertyName;
+
+    public function __construct(TokenStorageInterface $tokenStorage = null, PropertyAccessorInterface $propertyAccessor = null, string $propertyName = 'username')
```
Comment (nclsHart) [inline src/Provider/User/SymfonyUserProvider.php]: I think we should inverse `$propertyAccessor` and `$propertyName`.

File: src/MonologExtraBundle.php [hunk 6]
```diff
@@ -4,6 +4,6 @@
 
 use Symfony\Component\HttpKernel\Bundle\Bundle;
 
-class TheRedDotMonologExtraBundle extends Bundle
+class MonologExtraBundle extends Bundle
```
Comment (nclsHart) [inline src/MonologExtraBundle.php]: This change will introduce a BC break.

Maybe this should be extracted and reserved for 5.x.

**Review by nclsHart — CHANGES_REQUESTED** [review]

File: src/Provider/User/SymfonyUserProvider.php (hunk 3 above)
Comment (nclsHart) [reply src/Provider/User/SymfonyUserProvider.php]: Mark this const as private will introduce a BC break. But I think it's reasonable.

File: src/MonologExtraBundle.php (hunk 6 above)
Comment (notFloran) [reply src/MonologExtraBundle.php]: strange I do not remember making this change

**Comment by notFloran** [comment]
@nclsHart

**Review by nclsHart — APPROVED** [review]

**Pull request merged**